homeassistant/components/influxdb.py

```diff
@@ -33,8 +35,8 @@
 CONFIG_SCHEMA = vol.Schema({
     DOMAIN: vol.Schema({
         vol.Optional(CONF_HOST, default=DEFAULT_HOST): cv.string,
-        vol.Required(CONF_USERNAME): cv.string,
-        vol.Required(CONF_PASSWORD): cv.string,
+        vol.Optional(CONF_USERNAME, default=DEFAULT_USERNAME): cv.string,
+        vol.Optional(CONF_PASSWORD, default=DEFAULT_PASSWORD): cv.string,
```

Comment:
There are no defaults needed. `username = conf.get(CONF_USERNAME)` and password will be `None` if they are missing in the configuration.

We should also ensure that both (username and password) are present (`vol.Exclusive` / `vol.Inclusive`) if a password is given. As far as I remember is it possible to use only a username without password but I'm not sure.